### DISCUSSION TITLE
nextjs hybrid app for large app

### DISCUSSION THREAD

**Discussion by nohanmohsin** [vercel/next.js#41296] [Ideas]
I am making a sort of skillshare clone that has an added feature of notes where educators can share notes with their students. There are certain pages or features in this website that need ssr and some that I dont feel ssr is good for(eg: the notes feature). Should I consider using nextjs for this hybrid or am I better off just making it in csr

**Comment by icyJoseph** [comment]
A bit of simplified statement, but Next.js is a framework that can do both. Often SSR is understood in the classic sense that a page's HTML and all other assets are purely made on the server and shipped to the client, but modern SSR frameworks are actually doing things differently.

Take for example getServerSideProps, which makes a page be rendered on demand. If you arrive to this page from an external domain, Next.js renders the page client side and sends an HTML document to the client. However, if you are already at the application domain and navigate, through SPA transitions, into the getServerSideProps page, Next.js sends JSON and JS to the client, the JS is used to render the page, and JSON is used as props, isn't this CSR?

There's a lot of nuances when it comes to SPA's, and how the HTML your visitor is interacting with has been generated.

So anything you can do with a CSR framework you can do by definition with Next.js, because it just gives you more tools, and control over your pages and the data they use.
